plugins/new-relic/retention.yaml

```diff
@@ -808,4 +808,206 @@ presetScripts:
+
+          # If remote_addr is a pod, get its name. If not, use IP address.
+          df.ra_pod = px.pod_id_to_pod_name(px.ip_to_pod_id(df.remote_addr))
+          df.is_ra_pod = df.ra_pod != ''
+          df.ra_name = px.select(df.is_ra_pod, df.ra_pod, df.remote_addr)
+
+          df.is_server_tracing = df.trace_role == 2
+          df.is_source_pod_type = px.select(df.is_server_tracing, df.is_ra_pod, True)
+          df.is_dest_pod_type = px.select(df.is_server_tracing, True, df.is_ra_pod)
+
+          # Set client and server based on trace_role.
+          df.source_pod = px.select(df.is_server_tracing, df.ra_name, df.pod)
+          df.destination_pod = px.select(df.is_server_tracing, df.pod, df.ra_name)
+
+          df = df.drop(['ra_pod', 'is_ra_pod', 'ra_name', 'is_server_tracing'])
+          df.source_service = px.pod_name_to_service_name(df.source_pod)
+          df.destination_service = px.pod_name_to_service_name(df.destination_pod)
+          df.destination_namespace = px.pod_name_to_namespace(df.destination_pod)
+          df.source_namespace = px.pod_name_to_namespace(df.source_pod)
+
+          df.source_service = remove_ns_prefix(df.source_service)
+          df.destination_service = remove_ns_prefix(df.destination_service)
+          df.source_pod = remove_ns_prefix(df.source_pod)
+          df.destination_pod = remove_ns_prefix(df.destination_pod)
+
+          # If the destination service is missing then try the nslookup
+          df.destination_service = px.select(
+              df.destination_service != '',
+              df.destination_service,
+              px.nslookup(df.remote_addr),
+          )
+          # If the destination service is still missing then set the remote_addr
+          df.destination_service = px.select(
+              df.destination_service != '',
+              df.destination_service,
+              df.remote_addr,
+          )
+          return df
+
+
+      df = px.DataFrame('redis_events', start_time=px.plugin.start_time, end_time=px.plugin.end_time)
+      df = add_source_dest_columns(df)
+
+      df.start_time = df.time_ - df.latency
+      df.latency = df.latency / (1000 * 1000)
+      df.req_bytes = px.length(df.req_args)
+      df.resp_bytes = px.length(df.resp)
+      df.pixie = 'pixie'
+      df.cluster_id = px.vizier_id()
+      df.cluster_name = px.vizier_name()
+
+      otel_df = df
+      px.export(otel_df, px.otel.Data(
+        resource={
+          'service.name': otel_df.source_service,
+          'service.instance.id': otel_df.source_pod,
+          'k8s.namespace.name': otel_df.source_namespace,
+          'redis.service.name': otel_df.destination_service,
+          'redis.pod.name': otel_df.destination_pod,
+          'redis.namespace.name': otel_df.destination_namespace,
+          'redis.req_cmd': otel_df.req_cmd,
+          'k8s.cluster.name': otel_df.cluster_name,
+          'px.cluster.id': otel_df.cluster_id,
+          'instrumentation.provider': otel_df.pixie,
```

Comment:
Should we add `df.db_system = 'redis'` to both scripts?

Reply:
db_system might be a specific catch all for the SQL nerdlet. redis wouldn't work properly there.